code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm

```diff
@@ -71,6 +71,12 @@
 /mob/living/simple_animal/hostile/megafauna/proc/spawn_crusher_loot()
 	loot = crusher_loot
 
+
+/mob/living/simple_animal/hostile/megafauna/Life()
+	if(z != ZLEVEL_MINING)
+		z = ZLEVEL_MINING  
+	..()
```

Comment:
This should be in beforeShuttleMove. Just override the proc and return NONE

Reply:
Well it may need to check for the shuttle dock, fucking up the shuttle if a megafauna gets to it though is acceptable.